### PR TITLE
GEODE-7920: Geode UDP INT thread found processing cache operations

### PR DESCRIPTION
**Pull request by bschuchardt**
Modified DistributionMessage to look for JGroups "internal" executor
threads.  We thought we'd turned off all JGroups thread pools but this
one is still around.  We don't want to process DistributionMessages in
these threads unless absolutely necessary since they're needed when
processing incoming messages.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionMessage.java [hunk 1]
```diff
@@ -476,13 +477,19 @@ protected boolean mayNotifySerialGatewaySender(ClusterDistributionManager dm) {
...
 
   /**
-   * returns true if the current thread should not be used for inline processing. i.e., it is a
-   * "precious" resource
+   * returns true if the current thread should not be used for inline processing because it
+   * is responsible for reading geode-membership messages. Blocking such a thread can cause
+   * a server to be kicked out
    */
-  public static boolean isPreciousThread() {
+  public static boolean isMembershipMessengerThread() {
```
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionMessage.java:484]: the old name was very creative, but a bit subjective... like the rename

File: geode-core/src/main/java/org/apache/geode/distributed/internal/ShutdownMessage.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ protected void process(final ClusterDistributionManager dm) {
     // reply.setRecipient(getSender());
```
Comment (echobravopapa) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/ShutdownMessage.java:79]: looks like some dead code cruft ... maybe we scrape it off while we're nearby

File: geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionMessageTest.java [hunk 3]
```diff
@@ -35,4 +39,12 @@ public void shouldBeMockable() throws Exception {
 
     verify(mockDistributionMessage, times(1)).setReplySender(mockReplySender);
   }
+
```
Comment (echobravopapa) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionMessageTest.java:42]: is/was there a failing test that caught the underlying issue?

Comment (bschuchardt) [reply geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionMessageTest.java:42]: No, this was in thread dumps when I was debugging an application with disable-tcp=true.

File: geode-core/src/main/java/org/apache/geode/distributed/internal/ShutdownMessage.java (hunk 2 above)
Comment (bschuchardt) [reply geode-core/src/main/java/org/apache/geode/distributed/internal/ShutdownMessage.java]: removed

**Pull request merged**